src/SecTester.Bus/Extensions/HttpResponseMessageExtensions.cs

```diff
@@ -0,0 +1,51 @@
+using System;
+using System.Net.Http;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+
+namespace SecTester.Bus.Extensions;
+
+internal static class HttpResponseMessageExtensions
+{
+  private const int ContentLengthThreshold = 4096;
+  private const string DetailsContentType = "text/html";
+  private const string StripHtmlTagsRegex = "<[^>]*>";
+
+  public static HttpResponseMessage VerifySuccessStatusCode(this HttpResponseMessage httpResponseMessage)
+  {
+    if (httpResponseMessage.CanHandle())
+    {
+      var message = httpResponseMessage.TryExtractMessage();
+
+      if (message is not null)
+      {
+        throw new HttpRequestException(
+          $"{message}: {(int)httpResponseMessage.StatusCode} ({httpResponseMessage.ReasonPhrase}).");
+      }
+    }
+
+    return httpResponseMessage.EnsureSuccessStatusCode();
+  }
+
+  private static bool CanHandle(this HttpResponseMessage httpResponseMessage)
+  {
+    return (int)httpResponseMessage.StatusCode >= 400 &&
+           httpResponseMessage.Content.Headers.ContentType is not null &&
+           httpResponseMessage.Content.Headers.ContentLength is > 0 and < ContentLengthThreshold &&
+           DetailsContentType.Equals(httpResponseMessage.Content.Headers.ContentType.MediaType,
+             StringComparison.OrdinalIgnoreCase);
```

Comment:
So, does it means you are not going to handle `text/plain`? It is not clear why don't use `IsSuccessStatusCode`. We have already discussed that we don't expect any other content-type

Reply:
At the moment the client receives `html/text`. `IsSuccessStatusCode` is `true` for  `2xx` falimy only.

Reply:
So? I can change it to text/plain and your whole logic will be wrong. 

> IsSuccessStatusCode is true for 2xx falimy only.

And this is fine